pallets/qa-tools/src/lib.rs

```diff
@@ -0,0 +1,191 @@
+// This file is part of the SORA network and Polkaswap app.
+
+// Copyright (c) 2020, 2021, Polka Biome Ltd. All rights reserved.
+// SPDX-License-Identifier: BSD-4-Clause
+
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+
+// Redistributions of source code must retain the above copyright notice, this list
+// of conditions and the following disclaimer.
+// Redistributions in binary form must reproduce the above copyright notice, this
+// list of conditions and the following disclaimer in the documentation and/or other
+// materials provided with the distribution.
+//
+// All advertising materials mentioning features or use of this software must display
+// the following acknowledgement: This product includes software developed by Polka Biome
+// Ltd., SORA, and Polkaswap.
+//
+// Neither the name of the Polka Biome Ltd. nor the names of its contributors may be used
+// to endorse or promote products derived from this software without specific prior written permission.
+
+// THIS SOFTWARE IS PROVIDED BY Polka Biome Ltd. AS IS AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL Polka Biome Ltd. BE LIABLE FOR ANY
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+// BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+// private-net to make circular dependencies work
+#[cfg(all(test, feature = "private-net", feature = "ready-to-test"))] // order-book
+mod tests;
+pub mod weights;
+pub use weights::*;
+mod pallets;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use common::{
+        AssetInfoProvider, AssetName, AssetSymbol, BalancePrecision, ContentSource, Description,
+    };
+    use frame_support::dispatch::DispatchErrorWithPostInfo;
+    use frame_support::traits::Get;
+    use frame_support::{dispatch::PostDispatchInfo, pallet_prelude::*};
+    use frame_system::pallet_prelude::*;
+    use order_book::{MomentOf, OrderBookId};
+    use sp_std::prelude::*;
+
+    #[pallet::pallet]
+    pub struct Pallet<T>(_);
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config + order_book::Config + trading_pair::Config {
+        type WeightInfo: WeightInfo;
+        type OrderBookOrderLifespan: Get<MomentOf<Self>>;
```

Comment:
`order_book::Config` trait already contains the same constant, why you need to duplicate it?

Reply:
Constants in `order_book` specify boundaries for the order lifespans, but this one dictates lifetimes of orders created with this pallet (value instead of the allowed range)

Now that I think about it, maybe we should specify the lifetime in extrinsic call instead.. @wer1st what do you think?

Reply:
Yes, we could move it to extrinsics

Reply:
Ok, will do